proactive-jupyter-kernel/kernel.py

```diff
@@ -1431,6 +1434,62 @@ def __set_generic_information_from_input_data__(self, proactive_common, input_da
             for gen_info in input_data['generic_info']:
                 proactive_common.addGenericInformation(gen_info[0], gen_info[1])
 
+    def __set_job_nodesource__(self, proactive_common, input_data):
+        if 'nodesource' in input_data:
+            self.__kernel_print_ok_message__('Adding node source information ...\n')
+            NODE_SOURCE_NAME = input_data['nodesource']
+            proactive_common.addGenericInformation("NODESOURCENAME", NODE_SOURCE_NAME)
+            from string import Template
+            selection_script_template = Template("""
+selected = false
+NODE_SOURCE_NAME = "$NODE_SOURCE_NAME"
+if (NODE_SOURCE_NAME) {
+    selected = (NODE_SOURCE_NAME.equals(System.getProperty("proactive.node.nodesource")));
+} else {
+    selected = true
+}
```

Comment:
Can the template scripts be saved in a different location than in kernel.py? 
Could we have resources files to store them?

Reply:
I am not sure, as this script is standard and not changed along the past 10 years, so this is why i put it in the code.

Reply:
I know this is not a good practice, but we can refactor later. We can create a template folder to store those scripts as template and load them in the code. But we need to test this concept before to not break anything.